test/h5test.h

```diff
@@ -78,8 +78,16 @@ extern pthread_key_t test_thread_info_key_g;
 /* Whether the tests are configured to concurrently execute in more than one thread */
 #define TEST_EXECUTION_CONCURRENT (GetTestMaxNumThreads() > 1)
 
-#define IS_MAIN_TEST_THREAD (!TEST_EXECUTION_CONCURRENT ||\
-    ((pthread_getspecific(test_thread_info_key_g)) && (((thread_info_t*)pthread_getspecific(test_thread_info_key_g))->thread_idx == 0)))
+#define IS_MAIN_TEST_THREAD (                                                                                \
```

Comment:
This macro needed to be revised in order to not suppress output when the `-maxthreads` option is specified. If the thread-specific key is not set, we have to assume that this is the main test thread as there isn't a good way to check otherwise (unless we use the library's existing threading mechanisms which provide a thread ID).